src/doom/g_game.c

```diff
@@ -972,7 +972,31 @@ static void SetMouseButtons(unsigned int buttons_mask)
 // Get info needed to make ticcmd_ts for the players.
 // 
 boolean G_Responder (event_t* ev) 
-{ 
+{
+    // [crispy] demo fast-forward
+    if (ev->type == ev_keydown && ev->data1 == key_demospeed && 
+        (demoplayback || gamestate == GS_DEMOSCREEN))
+    {
+        if (singletics == false)
```

Comment:
Somehow complicated way to write `singletics = !singletics`? Or is there more to it?